src/popup/router/components/RadioButton.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <label
+    class="radio-container"
+    :class="{ disabled }"
+  >
+    <input
+      :disabled="disabled"
+      :value="value"
+      :type="type"
+      :name="name"
+      @change="handleChange"
+    >
+    <span
+      class="radio-dot"
+      :class="{ checked: value }"
+    />
+    <slot class="radio-holder" />
+  </label>
+</template>
+
+<script>
+export default {
+  props: {
+    value: { type: [String, Number, Boolean], default: '' },
+    type: { type: String, default: 'checkbox' },
+    name: { type: String, default: '' },
+    disabled: { type: Boolean },
+  },
+  methods: {
+    handleChange(event) {
+      if (this.disabled) return;
+      this.$emit('input', event.target.checked);
+    },
+  },
+};
+</script>
+
+<style lang="scss" scoped>
+@use '../../../styles/variables';
+@use '../../../styles/mixins';
+
+.radio-container {
+  @include mixins.flex;
+
+  cursor: pointer;
+
+  input {
+    position: absolute;
+    opacity: 0;
+    height: 0;
+    width: 0;
+  }
+
+  &.disabled {
+    opacity: 50%;
+    cursor: not-allowed;
+  }
+
+  .radio-dot {
+    border-radius: 50%;
+    border: 1px solid rgba(255, 255, 255, 0.5);
+    width: 20px;
+    height: 20px;
+    background: variables.$color-bg-3;
+    margin-right: 6px;
+    position: relative;
```

Comment:
I suggest to organise CSS properties according to commonly used methodology: https://9elements.com/css-rule-order/